### DISCUSSION TITLE
Completions

### DISCUSSION THREAD

**Discussion by PurpleBooth** [ScoopInstaller/Scoop#4803] [Help and Questions] [answered]
Hey, 

thinking about packaging my apps with scoop, currently I only provide the binary for windows and homebrew on linux and windows. Scoop looks like the nicest option for windows.

What's a good example to look at for setting up bash/pwsh/zsh/fish completions?

**Comment by niheaven** [comment]
Currently no Windows auto completions, scoop only extracts archive and add bin path.

**Comment by rashil2000** [comment]
You can add a note on how users can source the completion files themselves, but as such there's no centralised place for completion files.

**Comment by rashil2000** [comment, marked as answer]
Actually, I think it's trivially easy to add completion sources support to Scoop. We can track it here: https://github.com/ScoopInstaller/Scoop/issues/3247

We can create a folder `~\scoop\completions`, and inside that, subfolders for `bash`, `zsh`, `pwsh`, `fish` etc. During installation, we can copy some specified files to this folder, and delete them during uninstallation.

**Reply by rashil2000** [reply]
In the JSON manifest, an object field can be added, something like this:

```json
...
    "completions": {
        "fish": "path\\to\\fish\\file.fish",
        "bash": "path\\to\\bash\\file.sh",
        "pwsh": "path\\to\\pwsh\\file.ps1"
    }
...
```

**Reply by niheaven** [reply]
First add a completion likes:

```json
"completions": "xxx.ps1"
```

I'm wondering where to put this file and if zsh/fish files are specified, how to use them or how to treat them in a PowerShell environment?

**Reply by rashil2000** [reply]
The sourcing part will have to be done manually. Users will have to add these lines to .bashrc/profile.ps1:

```bash
source ~/scoop/completions/bash/*.sh
```

```pwsh
Get-ChildItem "~/scoop/completions/*.ps1" | ForEach-Object { . $_ }
```
